### PR TITLE
Drools 5472 - Implement ExpressionFunctionUtils

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @mariofusco @jiripetrlik

see https://issues.redhat.com/browse/DROOLS-5472

1) Moved EXPRESSION-specific code to ExpressionFunctionUtils
2) Removed "method arity" management fomr CommonCodegenUtils
3) Add/updated tests

### CODE REVIEW

**Review by danielezonca — APPROVED** [review]

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    @Test
+    public void getConstantExpressionMethodDeclaration() {
+        Constant constant = new Constant();
+        constant.setDataType(DataType.DOUBLE);
+        constant.setValue(34.6);
+        int methodArity = new Random().nextInt(20);
+        String methodName = String.format(METHOD_NAME_TEMPLATE, constant.getClass().getSimpleName(), methodArity);
+        MethodDeclaration retrieved = ExpressionFunctionUtils.getConstantExpressionMethodDeclaration(methodName, constant, Collections.emptyList());
+        commonValidateConstant(retrieved, constant, methodName, double.class.getName());
+        //
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java:55]: Maybe you forget to describe something here?

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        int methodArity = new Random().nextInt(20);
+        String fieldName = "FIELD_NAME";
+        FieldRef fieldRef = new FieldRef(FieldName.create(fieldName));
+        String methodName = String.format(METHOD_NAME_TEMPLATE, fieldRef.getClass().getSimpleName(), methodArity);
+        MethodDeclaration retrieved = ExpressionFunctionUtils.getFieldRefExpressionMethodDeclaration(methodName, fieldRef, Collections.singletonList(getTypedClassOrInterfaceType(List.class.getName(), Collections.singletonList(KiePMMLNameValue.class.getName()))));
+        String expected = String.format("return kiePMMLNameValue.map(%1$s::getValue).orElse(%2$s);",
+                                        KiePMMLNameValue.class.getName(),
+                                        fieldRef.getMapMissingTo());
+        commonValidateFieldRef(retrieved, fieldRef, methodName, expected);
+        //
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java:76]: Again empty comment.

**Review by jiripetrlik — APPROVED** [review]
I have only two minor comments.

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java:55]: hi @jiripetrlik 
When methods tests two different conditions I use "//" to separate them

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/java/org/kie/pmml/compiler/commons/utils/ExpressionFunctionUtilsTest.java:76]: @jiripetrlik 
see above

**Pull request merged**